tests/test_config.py

```diff
@@ -269,3 +269,27 @@ def test_style_in_toml(self, monkeypatch: pytest.MonkeyPatch) -> None:
         monkeypatch.chdir(CONFIG_DATA / "valid_toml_numpydoc_pep257")
         run = _Run(["test_package"])
         assert run.config.style == ["numpydoc", "pep257"]
+
+    def test_boolopt_in_toml(self, monkeypatch: pytest.MonkeyPatch) -> None:
+        """Test that arguments of type BooleanOptionalAction work in toml files."""
+        monkeypatch.chdir(CONFIG_DATA / "valid_toml_boolopt")
+        run = _Run(["test_package"])
+        assert not run.config.summary_quotes_same_line
+
+        # dashes in the name dont allow the dot notation to get this value
```

Comment:
```suggestion
        # dashes in the name dont allow the dot notation to get this value
```

Can't you use `_` like on L277?

Reply:
They are registered differently ... cannot be done. I am not sure if there is any other less hacky way of getting that attribute.

This is what i get if i start a debugger in line 278
```python
(pdb) p run.config.__dict__.keys()
dict_keys([..., 'max_summary_lines', ... 'numpydoc-section-order', ...])
(pdb) p run.config.strip_whitespace
*** AttributeError: 'Namespace' object has no attribute 'strip_whitespace'
```

Reply:
Ah, you're finding so many issues with the current code 😅 

I think this is because of the way

Reply:
We can definitely note this and make an issue out of it (I don't feel like fixing it should be in the scope of this PR)

Reply:
I already did 😄 

See #167